Source/WebCore/Modules/url-pattern/URLPatternParser.cpp

```diff
@@ -330,5 +329,219 @@ String escapeRegexString(StringView input)
     return escapeRegexStringForCharacters(input.span16());
 }
 
+// https://urlpattern.spec.whatwg.org/#convert-a-modifier-to-a-string
+String convertModifierToString(Modifier modifier)
+{
+    switch (modifier) {
+    case Modifier::ZeroOrMore:
+        return "*"_s;
+    case Modifier::Optional:
+        return "?"_s;
+    case Modifier::OneOrMore:
+        return "+"_s;
+    default:
+        return { };
+    }
+}
+
+// https://urlpattern.spec.whatwg.org/#generate-a-regular-expression-and-name-list
+std::pair<String, Vector<String>> generateRegexAndNameList(const Vector<Part>& partList, const URLPatternStringOptions& options)
+{
+    StringBuilder result;
+    result.append("^"_s);
+
+    Vector<String> nameList;
+
+    for (const Part& part : partList) {
+        if (part.type == PartType::FixedText) {
+            if (part.modifier == Modifier::None)
+                result.append(escapeRegexString(part.value));
+            else
+                result.append(makeString("(?:"_s, escapeRegexString(part.value), ")"_s, convertModifierToString(part.modifier)));
+
+            continue;
+        }
+
+        ASSERT(!part.name.isEmpty());
+
+        nameList.append(part.name);
+
+        StringView regexpValue = part.value;
+
+        if (part.type == PartType::SegmentWildcard)
+            regexpValue = generateSegmentWildcardRegexp(options);
```

Comment:
This cannot be safe. This is likely returning a newly constructed String, then you're constructing a StringView from it, then the temporary String gets destroyed and your StringView points to stale memory. Clear security bug.